components/arm/client.go

```diff
@@ -13,33 +13,42 @@ import (
 
 	rprotoutils "go.viam.com/rdk/protoutils"
 	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/resource"
 	"go.viam.com/rdk/spatialmath"
 )
 
 var errArmClientModelNotValid = errors.New("unable to retrieve a valid arm model from arm client")
 
 // client implements ArmServiceClient.
 type client struct {
+	resource.Named
+	resource.TriviallyReconfigurable
 	name   string
-	conn   rpc.ClientConn
 	client pb.ArmServiceClient
 	model  referenceframe.Model
 	logger golog.Logger
 }
 
 // NewClientFromConn constructs a new Client from connection passed in.
-func NewClientFromConn(ctx context.Context, conn rpc.ClientConn, name string, logger golog.Logger) Arm {
-	c := pb.NewArmServiceClient(conn)
-	// TODO: DATA-853 requires that this support models being changed on the fly, not just at creation
-	// TODO: RSDK-882 will update this so that this is not necessary
+func NewClientFromConn(ctx context.Context, conn rpc.ClientConn, name resource.Name, logger golog.Logger) (Arm, error) {
+	pbClient := pb.NewArmServiceClient(conn)
+	// TODO(DATA-853): requires that this support models being changed on the fly, not just at creation
+	// TODO(RSDK-882): will update this so that this is not necessary
```

Comment:
@spieswl theses TODOs are for closed tickets. Is there work we need to do here, or should the TODOs be removed?

Reply:
- L35's comment relating to DATA-853 can be removed. This was handled in a later ticket.
- L36's comment on RSDK-882 is still valid; this comment is specifically about getting the `model` from the `RobotServiceClient` but should be addressed by the planned `GetGeometry` changes. We could update the ticket to reference RSDK-2539, which is the epic associated with `GetGeometry`